plugins/outputs/zabbix/autoregister_test.go

```diff
@@ -0,0 +1,123 @@
+package zabbix
+
+import (
+	"testing"
+	"time"
+
+	"github.com/datadope-io/go-zabbix/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+// TestZabbixAutoregisterDisabledAdd tests that Add does not store information if autoregister is disabled.
+func TestZabbixAutoregisterDisabledAdd(t *testing.T) {
+	z := Zabbix{
+		autoregisterLastSend: make(map[string]time.Time),
+	}
+
+	z.autoregisterAdd("hostname")
+	require.Equal(t, len(z.autoregisterLastSend), 0)
+}
+
+// TestZabbixAutoregisterEnabledAdd tests that Add stores information if autoregister is enabled.
+func TestZabbixAutoregisterEnabledAdd(t *testing.T) {
+	z := Zabbix{
+		Autoregister:         "autoregister",
+		autoregisterLastSend: make(map[string]time.Time),
+	}
+
+	z.autoregisterAdd("hostname")
+	require.Equal(t, len(z.autoregisterLastSend), 1)
+
+	_, exists := z.autoregisterLastSend["hostname"]
+	require.True(t, exists)
+}
+
+// TestZabbixAutoregisterPush tests different cases of Push with a table oriented test.
+func TestZabbixAutoregisterPush(t *testing.T) {
+	zabbixSender := &mockZabbixSender{}
+	z := Zabbix{
+		Log:                        testutil.Logger{},
+		AutoregisterResendInterval: config.Duration(1 * time.Second),
+		autoregisterLastSend:       make(map[string]time.Time),
+		sender:                     zabbixSender,
+	}
+
+	// Test that nothing is sent if autoregister is disabled.
+	z.autoregisterPush()
+	require.Equal(t, len(z.autoregisterLastSend), 0)
+
+	// Test that nothing is sent if autoregister is enabled but no host is added.
+	z.Autoregister = "autoregister"
+	z.autoregisterPush()
+	require.Equal(t, len(z.autoregisterLastSend), 0)
+
+	// Test that autoregister is sent if autoregister is enabled and a host is added.
+	z.Autoregister = "autoregister"
+	z.autoregisterAdd("hostname")
+	z.autoregisterPush()
+	require.Equal(t, len(z.autoregisterLastSend), 1)
+	require.Equal(t, zabbixSender.hostname, "hostname")
+	require.Equal(t, zabbixSender.hostMetadata, "autoregister")
```

Comment:
Expected comes first
```suggestion
	require.Equal(t, "hostname", zabbixSender.hostname)
	require.Equal(t, "autoregister", zabbixSender.hostMetadata)
```